### ISSUE TITLE
Block Settings Sidebar refresh when we change the Editor responsive Preview.

### ISSUE THREAD

**Issue by RajkiranBagal** [WordPress/gutenberg#38667]
### Description

When we change the Preview from one device type to another in the Block editor then the block settings sidebar gets refreshed.

### Step-by-step reproduction instructions

1. Drag & Drop the Heading Block in the Editor.
2. Go to settings, set some color, or anything.
3. Now click on the Preview button & select Tablet - ( https://share.bsf.io/nOu97DnJ )
4. You can see the Heading Block settings sidebar gets refreshed & now you again need to open the Panel & find the setting you wanted to change.

### Screenshots, screen recording, code snippet

Screencast - https://share.bsf.io/KouYklQ0

### Environment info

Not installed. Just using WordPress version 5.9

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by TimothyBJacobs** [comment]
I've also seen this behavior. It is a regression from 5.8. I believe this is because device previews are now rendered in an iFrame as of 5.9. It looks like the `children` internal component state is lost when they are rendered into an iFrame.

This can cause a frustrating UX when trying to adjust a particular setting while checking how it looks in each viewport.

Is there a way that we can maintain the component's state here? Or is this only fixable by always rendering into an iFrame.

**Comment by andreiglingeanu** [comment]
Noticed that too. A very annoying issue that unfortunately cannot be fixed just from the components inside the block settings sidebar (only if we fully extract the state outside the components rendered here).

Also, another issue is, moving from `desktop` to `tablet` device feels very laggy due to the initial render of the iframe.

**Comment by 2xSamurai** [comment]
Confirming that it's still there and really annoying since the UX is ruined. It happens when the preview is changed from  'Desktop' to **Tablet / Mobile** or back, ie when the iframe is introduced or removed. Switching between **Mobile** and **Tablet** seems to be okay and doesn't reload/reset **InspectorControls** states.

**Comment by mark-c-woodard** [comment]
Something I noticed that is potentially related: 
When you have blocks inside an `<EntityProvider>` and you change the editor preview, the blocks' clientIds will change / refresh.

This doesn't happen when all blocks are on blocks api 3. So it appears to be an issue with the switch from non-iframed to iframed preview.

**Comment by xdrew760x** [comment]
Exactly, i just came across this issue when working with different attributes for different devicetype selected. I noticed when jumping from tablet to mobile it wouldn't re render the options it kept tablets. This is a result of this issue you are all talking about as well. Im currently looking for some kind of fix so my users don't need to jump between desktop each time they want to update how the mobile will look.

**Comment by Mamaduka** [comment]
Like @mark-c-woodard, noted this doesn't happen with iframed editors. Currently, WP will disable the iframed editor when some blocks are using older API versions, but the plan is to remove this condition. See: #70743.

cc @t-hamano 

P.S. Related closed issue - #38258.

**Comment by t-hamano** [comment]
I propose to close this issue as "won't fix."

The desktop view disables the iframe editor based on several conditions. On the other hand, the tablet and mobile view always works as iframed editors. This difference causes a rerender, which is difficult to solve.

When the desktop preview becomes to work as the iframe editor in the future, the problems reported here will be automatically resolved.

**Comment by Mamaduka** [comment]
Thanks for the feedback, @t-hamano!

Let's close this as "won't fix".
